apm-sniffer/apm-sdk-plugin/grpc-1.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/grpc/v1/StreamCallClientInterceptor.java

```diff
@@ -95,18 +95,25 @@ protected CallListener(Listener delegate, ContextSnapshot contextSnapshot) {
             this.contextSnapshot = contextSnapshot;
         }
 
-        @Override public void onReady() {
+        @Override
+        public void onReady() {
             delegate().onReady();
         }
 
-        @Override public void onHeaders(Metadata headers) {
+        @Override
+        public void onHeaders(Metadata headers) {
             delegate().onHeaders(headers);
         }
 
-        @Override public void onMessage(Object message) {
+        @Override
+        public void onMessage(Object message) {
             try {
                 ContextManager.createLocalSpan(operationPrefix + STREAM_RESPONSE_OBSERVER_ON_NEXT_OPERATION_NAME);
                 ContextManager.continued(contextSnapshot);
+            } catch (Throwable t) {
+                // ignore trace error
```

Comment:
Why trace has error?

Reply:
ContextManager.continued has status check, It is maybe throws `IllegalArgumentException`, So using `try-catch` block avoid trace error break normal logic.

Reply:
No, you should not do that. You should check the `contextSnapshot != null`. Exception throw/catch is a heavy OP, it costs more CPU than you thought.